projects/optic/src/utils/write-optic-config.ts

```diff
@@ -0,0 +1,21 @@
+import yaml from 'yaml';
+import fs from 'fs';
+import { logger } from '../logger';
+
+// merges a yaml document into the existing optic.yml and writes it to disk. if newDocument's top-level
+// key already exists in the Optic config, it is overwritten.
+export function updateOpticConfig(
+  newDocument: yaml.Document.Parsed,
+  opticConfigPath: string
+) {
+  try {
+    let opticYml = yaml.parseDocument(fs.readFileSync(opticConfigPath, 'utf8'));
+
+    const topLevelKey = Object.keys(newDocument.toJSON())[0];
+    opticYml.set(topLevelKey, newDocument.get(topLevelKey));
+
+    fs.writeFileSync(opticConfigPath, opticYml.toString());
+  } catch (err) {
+    logger.error(err);
```

Comment:
Instead of logging this error here, we might want to let the error throw up to the top level and handle it in the `capture.ts` file